### PR TITLE
Python 3.12 tooling / CI support

### PR DESCRIPTION
**Pull request by SamuelMarks**
I saw 5f7f05a so I guess we're close to Python 3.12 support. Committing this up so I can see what passes/fails and fix accordingly.

### CODE REVIEW

**Comment by MichaelHudgins** [comment]
@vladbelit could you PTAL

File: .github/workflows/arm-cd.yml
Comment (MichaelHudgins) [inline .github/workflows/arm-cd.yml:1]: Please omit this file for now.

**Review by MichaelHudgins — CHANGES_REQUESTED** [review]

**Comment by vladbelit** [comment]
We have a similar change going out soon (nightly jobs), though there are still a couple of details being hashed out with regards to CI

**Comment by gbaned** [comment]
Hi @SamuelMarks Can you please rebase your branch and resolve conflicts? Thank you!

**Comment by vladbelit** [comment]
This could probably be closed - the Linux x86/ARM 3.12 nightlies are up. They should soon be set to upload, once a few superficial test failures are resolved.
MacOS should be added soon as well.

**Comment by MichaelHudgins** [comment]
Thanks for the PR @SamuelMarks but the internal effort landed and our continued development of the official directory would have needed this PR to change multiple times.

**Pull request closed without merging**